packages/eslint-plugin/src/rules/top-level-styles.md

```diff
@@ -0,0 +1,64 @@
+# Ensure `makeStyles`, `makeResetStyles` and `makeStaticStyles` are only called at the top-level of a file
+
+This is an opinionated rule to discourage developers from misusing the `makeStyles`, `makeResetStyles` and `makeStaticStyles` 
```

Comment:
```suggestion
This is a rule to discourage developers from misusing the `makeStyles`, `makeResetStyles` and `makeStaticStyles` 
```

Agree with @spmonahan, there are no opinions, `makeStyles()` & co. calls should be top-level 🐱